### PR TITLE
Fix friendly name and tests simple native phonon

### PR DESCRIPTION
**Pull request by npmaile**

### CODE REVIEW

File: Phonon/src/io/gridplus/phonon/PhononApplet.java [hunk 1]
```diff
@@ -836,16 +836,17 @@ private void MinePhonon( APDU apdu )
...
     	// Hash Value is ScratchBuffer used to determine rarity
-
-        crypto.sha512.update(ScratchBuffer, (short)1, (short)64);
-       	crypto.random.generateData(TransBuffer, (short) 0, (short) 64);
-       	crypto.sha512.update( TransBuffer, (short)0, (short)64);
-        crypto.sha512.doFinal(ScratchBuffer, (short) 0, (short)64, ScratchBuffer, (short) 0);
+	
+	// get rid of the pubkey for now for alpha
+	// todo: add back in the pubkey
+        //crypto.sha512.update(ScratchBuffer, (short)1, (short)64);
```
Comment (maevyn11) [inline Phonon/src/io/gridplus/phonon/PhononApplet.java:847]: I don't think we're ever going to put the pubkey back in. It's not additive to the security of the mined phonons and it's inclusion was just causing practical derivation and validation problems. So I think we should just delete this commented out code.

File: Phonon/src/io/gridplus/phonon/PhononApplet.java [hunk 2]
```diff
@@ -932,23 +933,21 @@ private void MinePhonon( APDU apdu )
 
         Util.arrayCopyNonAtomic(ScratchBuffer, (short) 0, PhononArray[phononKeyPointer].sPhononPublicKey, (short) 0, (short)64);
 
-        PhononArray[phononKeyPointer].PhononPrivateKeyLen = (short)64;
+        PhononArray[phononKeyPointer].PhononPrivateKeyLen = (short)32;
         if (UsingDeletedSpot == 0) {
             PhononArray[phononKeyPointer].sPhononPrivateKey = new byte[PhononArray[phononKeyPointer].PhononPrivateKeyLen];
         }
 
         // 64 byte of Salt ( TransBuffer ) is private key value
```
Comment (maevyn11) [inline Phonon/src/io/gridplus/phonon/PhononApplet.java:941]: should update comment to say 32 byte

File: Phonon/src/io/gridplus/phonon/PhononApplet.java [hunk 3]
```diff
@@ -1173,27 +1172,37 @@ private void ReceivePhonons(APDU apdu) {
...
+            	short PublicKeyLength = secp256k1.derivePublicKey(PrivateKey, PublicKeystr, (short) 0);
+            	ECPublicKey PublicKey = (ECPublicKey) PhononKey.getPublic();
+            	PublicKey.setW(PublicKeystr, (short) 0, PublicKeyLength);
+
+            	PhononArray[phononKeyPointer].PhononPublicKeyLen = PublicKeyLength;
+
+            	if (UsingDeletedSpot == 0) {
+                	PhononArray[phononKeyPointer].sPhononPublicKey = new byte[PhononArray[phononKeyPointer].PhononPublicKeyLen];
+            	}
+            	Util.arrayCopyNonAtomic(PublicKeystr, (short) 0, PhononArray[phononKeyPointer].sPhononPublicKey, (short) 0, PhononArray[phononKeyPointer].PhononPublicKeyLen);
```
Comment (maevyn11) [inline Phonon/src/io/gridplus/phonon/PhononApplet.java:1195]: Can see that this works empirically and I think I can trace the code, but damn this is a lot of steps to set a public key in the phonon table. We should def work on wrapping this sort of thing into separate functions post testnet.

File: Phonon/src/io/gridplus/phonon/PhononApplet.java [hunk 4]
```diff
@@ -1400,14 +1413,10 @@ private void SetPhononDescriptor(APDU apdu) {
...
-                ExtendedSchemaVersion == UNINITIALIZED_SHORT ||
-                ValueBaseSet == false ||
-                ValueExponentSet == false) {
-            secureChannel.respond(apdu, (short) 0, ISO7816.SW_DATA_INVALID);
-            return;
-        }
+	if (!ValueBaseSet || !ValueExponentSet || !ValueSchemaSet || !ValueExtendedSchemaSet || !ValueCurrencySet){
+		secureChannel.respond(apdu, (short) 0, ISO7816.SW_DATA_INVALID);
+		return;
+	}
```
Comment (maevyn11) [inline Phonon/src/io/gridplus/phonon/PhononApplet.java:1419]: This code is already in main? 

Not sure what's going on here, has something not been merged. This code is fine, but it's already in the main branch so not sure why this diff is showing up.

File: Phonon/src/io/gridplus/phonon/SecureChannel.java [hunk 5]
```diff
@@ -22,7 +22,7 @@
  */
 public class SecureChannel {
     public static final boolean SECURE_CHANNEL_DEBUG = false;
-    public static final boolean USE_CA_DEMO_KEY = false;
+    public static final boolean USE_CA_DEMO_KEY = true;
```
Comment (maevyn11) [inline Phonon/src/io/gridplus/phonon/SecureChannel.java:25]: Need to flip to false before merging.

**Review by maevyn11 — CHANGES_REQUESTED** [review]
Looks good mostly, just a couple cleanup things and probably a weird github display issue. Would like Mike's feedback on the pubkey derivation section before merging.

File: Phonon/src/io/gridplus/phonon/PhononApplet.java (hunk 2 above)
Comment (SecureElement) [reply Phonon/src/io/gridplus/phonon/PhononApplet.java:941]: where is the comment?

File: Phonon/src/io/gridplus/phonon/PhononApplet.java (hunk 4 above)
Comment (SecureElement) [reply Phonon/src/io/gridplus/phonon/PhononApplet.java:1419]: This is work Nate did

File: Phonon/src/io/gridplus/phonon/PhononApplet.java (hunk 1 above)
Comment (SecureElement) [reply Phonon/src/io/gridplus/phonon/PhononApplet.java:847]: I am fine with it.  This is about how you want it to work

File: Phonon/src/io/gridplus/phonon/PhononApplet.java (hunk 2 above)
Comment (maevyn11) [reply Phonon/src/io/gridplus/phonon/PhononApplet.java:935]: line 935 `// 64 byte of Salt ( TransBuffer ) is private key value`

File: Phonon/src/io/gridplus/phonon/PhononApplet.java (hunk 3 above)
Comment (npmaile) [reply Phonon/src/io/gridplus/phonon/PhononApplet.java:1189]: will do post testnet

File: Phonon/src/io/gridplus/phonon/PhononApplet.java (hunk 2 above)
Comment (npmaile) [reply Phonon/src/io/gridplus/phonon/PhononApplet.java]: changed.

File: Phonon/src/io/gridplus/phonon/PhononApplet.java (hunk 1 above)
Comment (npmaile) [reply Phonon/src/io/gridplus/phonon/PhononApplet.java]: deleted

**Pull request merged**